src/org/rascalmpl/semantics/dynamic/PathPart.java

```diff
@@ -37,9 +44,29 @@ public Result<IValue> interpret(IEvaluator<Result<IValue>> __eval) {
 
 			Result<IValue> pre = this.getPre().interpret(__eval);
 			Result<IValue> expr = this.getExpression().interpret(__eval);
-			Result<IValue> tail = this.getTail().interpret(__eval);
-
-			return pre.add(expr).add(tail);
+			// here we have to encode the string for us in the path part of a uri
+			// the trick is to use new File which does the encoding for us, in 
+			// a way that conforms to the current OS where we are running. So if someone
+			// is splicing a windows path here, with the slashes the other way around,
+			// they are first interpreted as path separators and not encoded as %slash
+			// However, first we need to map the expression result to string by using
+			// the `add` semantics of strings:
+			IString path = (IString) ResultFactory.makeResult(TF.stringType(), VF.string(""), __eval)
+				.add(expr).getValue();
+			
+			try {
+				// reuse our URI encoders here on the unencoded expression part 
+				URI tmp = URIUtil.create("x", "", "/" + path.getValue());
```

Comment:
Hmm, I was thinking we should deprecate the `URI` part of `URIUtil`,  can we do this using ISourceLocation?

Reply:
possibly, but we also want to deprecate the URI part of ISourceLocation right? We'd have to think about `getRawPath` additions to ISourceLocation, and friends.

Reply:
well, to go from URI to ISourceLocation is fine(ish), but I was thinking that maybe we should not write new URI code when we don't need to.

Reply:
ah, okay, yeah that seems hackish, but this might be the place this kind of hackish stuff is needed. Can you write a test that checks that it works on windows? (and keeps working in the future/compiler?)

Maybe we need a special flag to say, run this test only on windows.

Reply:
I don't think this has to do with Windows. This is pure strings and URIs. The word "path" triggers thinking about file paths, but we are really not in that part of the code. This interpolation and the `+` operator will do the same thing exactly, 100% ,whether on Windows or on Mac or on Linux, for the same input and the same code.